internal/kafka/command_topic_produce.go

```diff
@@ -635,11 +635,19 @@ func (c *command) initSchemaAndGetInfo(cmd *cobra.Command, topic, mode string) (
 	}
 
 	// Fetch the SR client endpoint during schema registration
-	srClusterId, srEndpoint, err := c.GetCurrentSchemaRegistryClusterIdAndEndpoint()
+	srEndpoint, err := cmd.Flags().GetString("schema-registry-endpoint")
 	if err != nil {
 		return nil, nil, err
 	}
 
+	var srClusterId string
+	if (schemaId.IsSet() || schema != "") && srEndpoint == "" {
+		srClusterId, srEndpoint, err = c.GetCurrentSchemaRegistryClusterIdAndEndpoint()
+		if err != nil {
+			return nil, nil, err
+		}
+	}
+
```

Comment:
Same here, will the basic produce have issues with the empty `srClusterId` if `srEndpoint` is passed from customer flag?